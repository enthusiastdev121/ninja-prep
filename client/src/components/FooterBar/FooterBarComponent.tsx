import React, {Fragment} from 'react';
import FadeIn from 'react-fade-in';

import Responsiveness from 'utils/hocs/Responsiveness';

import FooterBarDesktop from './desktop/FooterBarDesktop';
import FooterBarMobile from './mobile/FooterBarMobile';

const desktop = <FooterBarDesktop />;
const mobile = <FooterBarMobile />;

interface Props {
  isLoadingUser: boolean;
}

const FooterBarComponent = (props: Props): JSX.Element => {
  if (props.isLoadingUser) {
    return <Fragment />;
  }
  return (
    <FadeIn>
      <Responsiveness desktop={desktop} mobile={mobile} />
    </FadeIn>
  );
};

export default FooterBarComponent;
