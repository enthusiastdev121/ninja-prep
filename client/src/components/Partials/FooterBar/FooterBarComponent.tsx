import React from 'react'
import FooterBarDesktop from './FooterBarDesktop'
import FooterBarTablet from './FooterBarTablet'
import FooterBarMobile from './FooterBarMobile'
import { Desktop, Tablet, Mobile } from '../../MediaQuery/DeviceSizes'

const FooterBarComponent = () => (
    <div>
        <Desktop>
            <FooterBarDesktop />
        </Desktop>
        <Tablet>
            <FooterBarTablet />
        </Tablet>
        <Mobile>
            <FooterBarMobile />
        </Mobile>
    </div>
)

export default FooterBarComponent
