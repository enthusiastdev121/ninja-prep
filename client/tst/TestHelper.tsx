import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
const history = createBrowserHistory()

interface DEVICE_WIDTHS {
    readonly DESKTOP: number
    readonly MOBILE: number
    readonly TABLET: number
}

//Widths in Pixels. Aligns with Media Query widths
export const DEVICE_TYPES: DEVICE_WIDTHS = {
    DESKTOP: 1000,
    MOBILE: 750,
    TABLET: 800
}

export const routeWrapper = (children: React.ReactChild) => {
    return <Router history={history}>{children}</Router>
}
