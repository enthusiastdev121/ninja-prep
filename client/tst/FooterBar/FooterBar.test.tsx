import React from 'react';

import FooterBarComponent from 'components/FooterBar';
import {mount, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Context as ResponsiveContext} from 'react-responsive';

import {DEVICE_TYPES, routeWrapper} from '../TestHelper';

test.each(Object.entries(DEVICE_TYPES))(
  'renders successfully on %p',
  (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider;
    const wrappingComponentProps = {value: {width: deviceWidth}};
    const mountProps = {wrappingComponent, wrappingComponentProps};

    const wrapper = shallow(routeWrapper(<FooterBarComponent />), mountProps);
    expect(wrapper).toBeDefined();
  },
);

test.each(Object.entries(DEVICE_TYPES))(
  'matches snapshot on %p',
  (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider;
    const wrappingComponentProps = {value: {width: deviceWidth}};
    const mountProps = {wrappingComponent, wrappingComponentProps};

    const wrapper = mount(routeWrapper(<FooterBarComponent />), mountProps);
    expect(toJson(wrapper)).toMatchSnapshot();
  },
);
