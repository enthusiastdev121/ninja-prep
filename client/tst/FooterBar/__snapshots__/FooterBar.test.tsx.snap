// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FooterBarComponent />
  </Router>
</Provider>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FooterBarComponent />
  </Router>
</Provider>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FooterBarComponent />
  </Router>
</Provider>
`;
